packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -109,7 +116,105 @@ export default createRule<
     const seenLogicals = new Set<TSESTree.LogicalExpression>();
 
     return {
-      // specific handling for `(foo ?? {}).bar` / `(foo || {}).bar`
+      // specific handling for `if (foo) { foo.bar(); }` / `if (foo) foo.bar();`
+      'IfStatement[consequent.type=BlockStatement][consequent.body.length=1], IfStatement[consequent.type=ExpressionStatement]':
+        (
+          node: {
+            consequent: {
+              type:
+                | AST_NODE_TYPES.BlockStatement
+                | AST_NODE_TYPES.ExpressionStatement;
+            };
+          } & TSESTree.IfStatement,
+        ): void => {
+          if (!options.allowSuggestingOnIfStatements) {
+            return;
+          }
+          if (node.alternate) {
+            return;
+          }
+          const ifBodyStatement =
+            node.consequent.type === AST_NODE_TYPES.BlockStatement
+              ? node.consequent.body[0]
+              : node.consequent;
+
+          if (ifBodyStatement.type !== AST_NODE_TYPES.ExpressionStatement) {
+            return;
+          }
+
+          const hasCommentsInIfBody =
+            context.sourceCode.getCommentsBefore(ifBodyStatement).length ||
+            context.sourceCode.getCommentsAfter(ifBodyStatement).length;
+
+          if (hasCommentsInIfBody) {
+            return;
+          }
```

Comment:
[Bug] I think we should still report in this case. Folks put comments all over the place sometimes; on their own they don't indicate a lint rule should be ignored.

```suggestion
```

Reply:
Looking through https://typescript-eslint.io/rules/?=fixable, the first few that jump to mind are:

* [@typescript-eslint/consistent-type-definitions](https://typescript-eslint.io/rules/consistent-type-definitions)
* [@typescript-eslint/method-signature-style](https://typescript-eslint.io/rules/method-signature-style)
* [@typescript-eslint/prefer-function-type](https://typescript-eslint.io/rules/prefer-function-type)

I wouldn't trust any existing rule to be honest. It's entirely possible there are existing bugs with comments. You might end up filing separate issues on other rules out of this 😄